### PR TITLE
Handle game error

### PR DESCRIPTION
**Pull request by VictorSchneuwly**
Handle the behavior of the game when a player crashed due to database issues.

### CODE REVIEW

File: app/src/androidTest/java/com/github/fribourgsdp/radio/unit/PlayerGameHandlerTest.kt [hunk 1]
```diff
@@ -15,9 +18,11 @@ import java.util.*
 
 
 class PlayerGameHandlerTest {
+    private val ctx: Context = ApplicationProvider.getApplicationContext()
+
     private lateinit var mockSnapshot: DocumentSnapshot
 
-    private val sleepingTime = 50L
+    private val sleepingTime = 500L
```
Comment (BartLucien) [inline app/src/androidTest/java/com/github/fribourgsdp/radio/unit/PlayerGameHandlerTest.kt:25]: Do you really need to increase the sleeping time?

File: app/src/main/java/com/github/fribourgsdp/radio/game/GameView.kt [hunk 2]
```diff
@@ -63,12 +63,12 @@ interface GameView: Timer.DeadlineHandler {
...
      * Handle the end of the game.
      */
-    fun gameOver(finalScores: Map<String, Long>)
+    fun gameOver(finalScores: Map<String, Long>?, hasCrashed: Boolean = false)
 
     /**
      * Displays the given [String] in the dedicated TextView
      */
-    fun displayLyrics(lyrics : String)
+    fun updateLyrics(lyrics : String)
```
Comment (BartLucien) [inline app/src/main/java/com/github/fribourgsdp/radio/game/GameView.kt:71]: I think you should update the documentation

File: app/src/main/java/com/github/fribourgsdp/radio/game/GameView.kt [hunk 3]
```diff
@@ -63,12 +63,12 @@ interface GameView: Timer.DeadlineHandler {
     /**
      * Handle the end of the game.
      */
-    fun gameOver(finalScores: Map<String, Long>)
+    fun gameOver(finalScores: Map<String, Long>?, hasCrashed: Boolean = false)
```
Comment (BartLucien) [inline app/src/main/java/com/github/fribourgsdp/radio/game/GameView.kt:66]: I think you should add the parameters in the doc.

File: app/src/androidTest/java/com/github/fribourgsdp/radio/unit/PlayerGameHandlerTest.kt (hunk 1 above)
Comment (BartLucien) [reply app/src/androidTest/java/com/github/fribourgsdp/radio/unit/PlayerGameHandlerTest.kt:25]: I think that even if it is for test of the same class, we can win some precious seconds by having the minimum time required to make the test run.

Comment (VictorSchneuwly) [reply app/src/androidTest/java/com/github/fribourgsdp/radio/unit/PlayerGameHandlerTest.kt]: So I put it back to 50L but I will see with @nathanduchesne why one of them was increased to 1000L.

File: app/src/main/java/com/github/fribourgsdp/radio/game/GameActivity.kt [hunk 4]
```diff
@@ -127,6 +137,12 @@ open class GameActivity : AppCompatActivity(), GameView, Timer.Listener {
     }
 
     override fun displayGuessInput() {
+        // Close the lyrics popup if already open
+        lyricsPopup?.let {
```
Comment (sim0mo) [inline app/src/main/java/com/github/fribourgsdp/radio/game/GameActivity.kt:141]: Minor : I think codeclimate missed something : This block is repeated 4 times

Comment (VictorSchneuwly) [reply app/src/main/java/com/github/fribourgsdp/radio/game/GameActivity.kt]: Done

**Review by sim0mo — APPROVED** [review]
Good code overall. You fixed what concerned the lyrics popup so I'm ok with it.

**Pull request merged**